src/app/components/budget-category-card/budget-category-card.component.ts

```diff
@@ -73,7 +74,30 @@ export class BudgetCategoryCardComponent implements AfterViewChecked {
         saveLineItemPayload.budgetCategoryId = this.budgetCategoryId;
 
         this.lineItemService.saveNewLineItem(saveLineItemPayload);
-        this.isAddingLineItem = false;
+        this.lineItemService.newlyAddedLineItemId
```

Comment:
Is this something that required rxjs specifically?

Reply:
Yeah, in this scenario it was best.

Reply:
it was an async operation hence had to use an observable